### ISSUE TITLE
scalajs-sbt-test-adapter: corrupt sources.jar

### ISSUE THREAD

**Issue by Andrei-Pozolotin** [scala-js/scala-js#3265]
somehow `sources.jar` vs `binary.jar` has different packages:

1. `sources` has `sbttestadapter`:
```
jar tf scalajs-sbt-test-adapter_2.12-0.6.21-sources.jar | sort
```
```
META-INF/MANIFEST.MF
org/
org/scalajs/
org/scalajs/sbttestadapter/
org/scalajs/sbttestadapter/ComJSEnvRPC.scala
org/scalajs/sbttestadapter/package.scala
org/scalajs/sbttestadapter/RemoteException.scala
org/scalajs/sbttestadapter/ScalaJSFramework.scala
org/scalajs/sbttestadapter/ScalaJSRunner.scala
org/scalajs/sbttestadapter/ScalaJSTask.scala
org/scalajs/testcommon/
org/scalajs/testcommon/Endpoints.scala
org/scalajs/testcommon/ExecuteRequest.scala
org/scalajs/testcommon/FrameworkInfo.scala
org/scalajs/testcommon/FrameworkMessage.scala
org/scalajs/testcommon/JSMasterEndpoints.scala
org/scalajs/testcommon/JSSlaveEndpoints.scala
org/scalajs/testcommon/JVMMasterEndpoints.scala
org/scalajs/testcommon/JVMSlaveEndpoints.scala
org/scalajs/testcommon/LogElement.scala
org/scalajs/testcommon/RPCCore.scala
org/scalajs/testcommon/RunnerArgs.scala
org/scalajs/testcommon/Serializer.scala
org/scalajs/testcommon/TaskInfo.scala

```
2. `binary` has `testadapter`:
```
jar tf scalajs-sbt-test-adapter_2.12-0.6.21.jar | sort
```
```
META-INF/MANIFEST.MF
org/
org/scalajs/
org/scalajs/testadapter/
org/scalajs/testadapter/ComJSEnvRPC$$anon$1.class
org/scalajs/testadapter/ComJSEnvRPC.class
org/scalajs/testadapter/package.class
org/scalajs/testadapter/package$.class
org/scalajs/testadapter/RemoteException.class
org/scalajs/testadapter/RemoteException$.class
org/scalajs/testadapter/RemoteException$Deserializer$.class
org/scalajs/testadapter/RemoteException$StackTraceDeserializer$.class
org/scalajs/testadapter/ScalaJSFramework.class
org/scalajs/testadapter/ScalaJSRunner.class
org/scalajs/testadapter/ScalaJSRunner$.class
org/scalajs/testadapter/ScalaJSRunner$RichFuture.class
org/scalajs/testadapter/ScalaJSRunner$RichFuture$.class
org/scalajs/testadapter/ScalaJSTask.class
org/scalajs/testadapter/ScalaJSTask$.class
org/scalajs/testadapter/ScalaJSTask$LogElement.class
org/scalajs/testcommon/
org/scalajs/testcommon/Endpoint.class
org/scalajs/testcommon/ExecuteRequest.class
org/scalajs/testcommon/ExecuteRequest$.class
org/scalajs/testcommon/ExecuteRequest$ExecuteRequestSerializer$.class
org/scalajs/testcommon/FrameworkInfo.class
org/scalajs/testcommon/FrameworkInfo$.class
org/scalajs/testcommon/FrameworkInfo$FrameworkInfoSerializer$.class
org/scalajs/testcommon/FrameworkMessage.class
org/scalajs/testcommon/FrameworkMessage$.class
org/scalajs/testcommon/FrameworkMessage$FrameworkMessageSerializer$.class
org/scalajs/testcommon/JSMasterEndpoints.class
org/scalajs/testcommon/JSMasterEndpoints$.class
org/scalajs/testcommon/JSSlaveEndpoints.class
org/scalajs/testcommon/JSSlaveEndpoints$.class
org/scalajs/testcommon/JVMMasterEndpoints.class
org/scalajs/testcommon/JVMMasterEndpoints$.class
org/scalajs/testcommon/JVMSlaveEndpoints.class
org/scalajs/testcommon/JVMSlaveEndpoints$.class
org/scalajs/testcommon/LogElement$$anon$1.class
org/scalajs/testcommon/LogElement.class
org/scalajs/testcommon/LogElement$.class
org/scalajs/testcommon/MsgEndpoint$$anon$1.class
org/scalajs/testcommon/MsgEndpoint.class
org/scalajs/testcommon/MsgEndpoint$.class
org/scalajs/testcommon/RPCCore$$anon$1.class
org/scalajs/testcommon/RPCCore$$anon$2.class
org/scalajs/testcommon/RPCCore$BoundEndpoint.class
org/scalajs/testcommon/RPCCore$BoundMsgEndpoint.class
org/scalajs/testcommon/RPCCore$BoundRPCEndpoint.class
org/scalajs/testcommon/RPCCore.class
org/scalajs/testcommon/RPCCore$.class
org/scalajs/testcommon/RPCCore$PendingCall$$anon$3.class
org/scalajs/testcommon/RPCCore$PendingCall.class
org/scalajs/testcommon/RPCCore$PendingCall$.class
org/scalajs/testcommon/RPCCore$RPCException.class
org/scalajs/testcommon/RPCEndpoint$$anon$2.class
org/scalajs/testcommon/RPCEndpoint.class
org/scalajs/testcommon/RPCEndpoint$.class
org/scalajs/testcommon/RunnerArgs.class
org/scalajs/testcommon/RunnerArgs$.class
org/scalajs/testcommon/RunnerArgs$RunnerArgsSerializer$.class
org/scalajs/testcommon/Serializer$$anon$4.class
org/scalajs/testcommon/Serializer$$anon$5.class
org/scalajs/testcommon/Serializer$BooleanSerializer$.class
org/scalajs/testcommon/Serializer$ByteSerializer$.class
org/scalajs/testcommon/Serializer.class
org/scalajs/testcommon/Serializer$.class
org/scalajs/testcommon/Serializer$DeserializeState.class
org/scalajs/testcommon/Serializer$DeserializeState$.class
org/scalajs/testcommon/Serializer$EventSerializer$$anon$6.class
org/scalajs/testcommon/Serializer$EventSerializer$.class
org/scalajs/testcommon/Serializer$FingerprintSerializer$$anon$2.class
org/scalajs/testcommon/Serializer$FingerprintSerializer$$anon$3.class
org/scalajs/testcommon/Serializer$FingerprintSerializer$.class
org/scalajs/testcommon/Serializer$IntSerializer$.class
org/scalajs/testcommon/Serializer$LongSerializer$.class
org/scalajs/testcommon/Serializer$OptionalThrowableSerializer$.class
org/scalajs/testcommon/Serializer$SelectorSerializer$.class
org/scalajs/testcommon/Serializer$SerializeState.class
org/scalajs/testcommon/Serializer$SerializeState$.class
org/scalajs/testcommon/Serializer$StackTraceElementSerializer$.class
org/scalajs/testcommon/Serializer$StatusSerializer$.class
org/scalajs/testcommon/Serializer$StringSerializer$.class
org/scalajs/testcommon/Serializer$TaskDefSerializer$.class
org/scalajs/testcommon/Serializer$ThrowableSerializer$$anon$1.class
org/scalajs/testcommon/Serializer$ThrowableSerializer$.class
org/scalajs/testcommon/Serializer$UnitSerializer$.class
org/scalajs/testcommon/TaskInfo.class
org/scalajs/testcommon/TaskInfo$.class
org/scalajs/testcommon/TaskInfo$TaskInfoSerializer$.class
```

**Comment by sjrd** [comment]
That's only because the directory structure does not match the package names, but they're still the correct sources. Is that causing a practical issue somehow?

Anyway it's already fixed in 1.x.

**Comment by Andrei-Pozolotin** [comment]
a bit annoying in Eclipse, for 0.6.x, if that version is going to be around for a long time

**Comment by sjrd** [comment]
I use Eclipse, including on the 0.6.x branch, and I do not notice any specific issue related to that. What exactly are the symptoms that you run into?

**Comment by sjrd** [comment]
I'm reluctant to move files around in the 0.6.x branch, because there have already been a lot of file moves in 1.x, and I am wary of whether git is going to follow along that well if there are some file moves in the two branches. I am afraid it might complicate future merges from 0.6.x into master. They are complicated enough as they currently stand.

**Comment by Andrei-Pozolotin** [comment]
ok, got it. then close. will use scala.js project as source reference

**Comment by sjrd** [comment]
OK, then let's consider this fixed in 1.0.0-M1 by cef9ae95602d49c1d3a6ff7208d88858f6c9aefc.
